### PR TITLE
Convert photorec to use RawDiskPartition

### PR DESCRIPTION
**Pull request by dfjxs**
Issue #78

### CODE REVIEW

**Comment by aarontp** [comment]
As you mentioned elsewhere, the previous changes are showing up here, so it's a bit hard to review.   I'm guessing you created this branch from the previous feature branch and not directly from master?   Maybe you could try syncing your master branch, then creating a new branch from there and then cherry-picking the actual new commits?  Maybe github is calculating the diffs based on the commits rather than the actual code diff here, and since those commits were squashed when merged into upstream master, it still thinks they are "new"?

**Comment by dfjxs** [comment]
Sorry about that, rebased the branch. Should be good now.

File: turbinia/workers/photorec.py [hunk 1]
```diff
@@ -44,16 +40,23 @@ def run(self, evidence, result):
     # Create the new Evidence object that will be generated by this Task.
     output_evidence = PhotorecOutput()
     # Create a path that we can write the new file to.
-    base_name = os.path.basename(evidence.local_path)
     output_file_path = os.path.join(self.output_dir, 'photorec_output')
     photorec_log = os.path.join(self.output_dir, 'photorec.log')
     # Add the output path to the evidence so we can automatically save it
     # later.
-    output_evidence.local_path = output_file_path
+    output_evidence.local_path = ''.join((output_file_path, '.1'))
```
[the remark this answers is not in the document]
Comment (dfjxs) [reply turbinia/workers/photorec.py:49]: Photorec appends .1 to whatever output folder specified.
I guess this is to support multiple runs with an incrementing number on the output folder.

**Comment by dfjxs** [comment]
@aarontp I've moved the extraction of volume extents into the partition enumeration task and stored the values in the evidence item.

I also changed the logic a little to support images that are taken at the volume level (i.e. no partition table).

**Review by aarontp — COMMENTED** [review]
Just a few small comments.

File: turbinia/processors/mount_local.py [hunk 2]
```diff
@@ -48,17 +50,18 @@ def PreprocessLosetup(source_path):
 
   if not os.path.exists(source_path):
     raise TurbiniaException(
-        'Cannot process non-existing source_path {0!s}'.format(source_path))
+        'Cannot mount non-existing source_path {0!s}'.format(source_path))
```
Comment (aarontp) [inline turbinia/processors/mount_local.py:53]: minor nit (sorry I missed this last time): We're not actually mounting it in this preprocessor.  Maybe 'Cannot create loopback for ..."?

File: turbinia/workers/photorec.py [hunk 3]
```diff
@@ -44,23 +42,32 @@ def run(self, evidence, result):
...
+      cmd = ' '.join(cmd)
       # Add a log line to the result that will be returned.
       result.log('Running photorec as [{0:s}]'.format(cmd))
       # Actually execute the binary
       self.execute(
           cmd, result, log_files=[photorec_log], new_evidence=[output_evidence],
           shell=True)
-      output_evidence.compress()
+      if os.path.exists(output_evidence.local_path):
+        output_evidence.compress()
```
Comment (aarontp) [inline turbinia/workers/photorec.py:70]: We should probably add a message here indicating that there was no output, and we might want to update the status message below to indicate that as well.

**Review by aarontp — COMMENTED** [review]
Just a couple minor things.

**Comment by dfjxs** [comment]
Fixed

**Review by aarontp — APPROVED** [review]
LGTM!

**Pull request merged**